### DISCUSSION TITLE
How to correctly get user's avatar URL?

### DISCUSSION THREAD

**Discussion by phenomen** [pocketbase/pocketbase#3217] [Q&A] [answered]
Here is my current code (Svelte). **It works** and I see both name and avatar, but I get a Typescript warning.

```html
<script lang="ts>
$: user = pb.authStore.model; //user: AuthModel
</script>

<img src={pb.files.getUrl(user, user.avatar)} alt={user.name}  />
```

A Typescript warning is at getUrl(**user**, user.avatar)
```
Argument of type 'RecordModel | AdminModel' is not assignable to parameter of type 'Pick<{ [key: string]: any; }, "id" | "collectionId" | "collectionName">'.
  Type 'AdminModel' is missing the following properties from type 'Pick<{ [key: string]: any; }, "id" | "collectionId" | "collectionName">': collectionId, collectionNamets(2345)

let user: RecordModel | AdminModel
```

**Comment by ganigeorgiev** [comment, marked as answer]
The warning is caused because `pb.authStore.model` is a union type (it can be both an auth record or admin).

I'll consider in the future loosening the `getFile()` types check (_and/or maybe add dedicated `user` and `admin` getters_), but for now you can try to:
- explicitly set the type of `user` when defining the variable to `RecordModel` (or type assert it; _to import package TS types you can do something like `import type { RecordModel } from 'pocketbase'`_) 
- explicitly set the type of `user` when defining the variable to `any` (or type assert it) 
- define a local user interface that contains the `id`, `collectionId` and `collectionName` fields.

**Reply by ganigeorgiev** [reply]
_Side-note: If the above code is for Svelte, please note that on ` pb.authStore.model` reassignment (it is actually a getter) the `user` variable will not be reactive._

**Reply by phenomen** [reply]
I think it will be reactive in my case since `pb` is reactive and its change will trigger `user` update (in $ if any right side variable changes, left side variable is updated). I omitted this code from the example since it's not relevant.
```typescript
export let data: LayoutData;

let { pb } = data;
$: ({ pb } = data);
$: user = pb.authStore.model as RecordModel; // this fixed type issue btw 
```

**Comment by heloineto** [comment]
I've created a helper function to safely get the avatar URL:

```ts
import type Pocketbase from 'pocketbase';
import type { Admin, Record } from 'pocketbase';

export function isAdmin(model: Record | Admin | null): model is Admin {
  return !!model && !(model as Record)?.collectionId;
}

export default function getAvatarUrl(
  pocketbase: Pocketbase,
  model: Record | Admin | null
) {
  if (!model) return undefined;
  if (isAdmin(model)) return undefined;
  if (typeof model.avatar !== 'string' || !model.avatar) return undefined;

  return pocketbase.getFileUrl(model, model.avatar);
}
```

This checks if it is safe to get the avatar, and makes typescript happy

**Reply by ganigeorgiev** [reply]
Based on the import types it seems that you are using an older JS SDK version. If you have the time, I'd recommend migrating to JS SDK v0.17+ to minimize the future breaking changes (_check the [v0.17.0 changelog](https://github.com/pocketbase/js-sdk/releases/tag/v0.17.0)_).
The `getUrl` check was also loosen with [JS SDK v0.17.3](https://github.com/pocketbase/js-sdk/releases/tag/v0.17.3) release.

**Reply by heloineto** [reply]
Thank you so much for letting me know! 🙌

Those are some great improvements! I'll update right away.
